samples/drivers/flash_shell/src/main.c

```diff
@@ -361,217 +372,202 @@ static bool page_layout_cb(const struct flash_pages_info *info, void *datav)
 	}
 
 	sz = info->size;
-	printk("\tPage %u: start 0x%08x, length 0x%lx (%lu, %lu KB)\n",
+	printk("\tPage %u: start 0x%08x, length 0x%lx (%lu, %lu KB)\r\n",
```

Comment:
you can extend struct page_layout_data to have shell handler and use PR_SHELL() here. Even though it is called callback, it is executed in the context of shell command so no worries about using shell_fprintf.